File: MTGSearch/src/test/java/com/dbottillo/mtgsearchfree/model/CardsBucketTest.java
package com.dbottillo.mtgsearchfree.model;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnit;
import org.mockito.junit.MockitoRule;

import java.util.Arrays;
import java.util.List;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertThat;
import static org.junit.Assert.assertTrue;
import static org.mockito.Mockito.when;

public class CardsBucketTest {

    @Rule
    public MockitoRule mockitoRule = MockitoJUnit.rule();

    @Mock
    MTGSet set;

    private CardsBucket cardsSetBucket;
    private CardsBucket genericBucket;
    private List<MTGCard> setCards = Arrays.asList(new MTGCard(5), new MTGCard(6));
    private List<MTGCard> genericCards = Arrays.asList(new MTGCard(8), new MTGCard(9));


    @Before
    public void setup() {
        when(set.getName()).thenReturn("Zendikar");
        cardsSetBucket = new CardsBucket(set, setCards);
        genericBucket = new CardsBucket("fav", genericCards);
    }

    @Test
    public void testGetCards() throws Exception {
        List<MTGCard> cards = cardsSetBucket.getCards();
        assertThat(cards, is(setCards));
        cards = genericBucket.getCards();
        assertThat(cards, is(genericCards));
    }

    @Test
    public void testIsValid() throws Exception {
        assertTrue(cardsSetBucket.isValid(set.getName()));
        assertFalse(cardsSetBucket.isValid("fav"));
        assertFalse(genericBucket.isValid(set.getName()));
        assertTrue(genericBucket.isValid("fav"));
    }

    @Test
    public void testSetCards() throws Exception {
        cardsSetBucket.setCards(genericCards);
        assertThat(cardsSetBucket.getCards(), is(genericCards));
    }

    @Test
    public void testGetKey() throws Exception {
        assertThat(cardsSetBucket.getKey(), is(set.getName()));
        assertThat(genericBucket.getKey(), is("fav"));
    }
}

File: MTGSearch/src/test/kotlin/com/dbottillo/mtgsearchfree/model/CardsBucketTest.kt
package com.dbottillo.mtgsearchfree.model

import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.*
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.mockito.Mock
import org.mockito.Mockito.`when`
import org.mockito.junit.MockitoJUnit
import java.util.*

class CardsBucketTest {

    @Rule
    @JvmField
    var mockitoRule = MockitoJUnit.rule()

    @Mock
    lateinit var set: MTGSet

    lateinit var cardsSetBucket: CardsBucket
    lateinit var genericBucket: CardsBucket

    private val setCards = Arrays.asList(MTGCard(5), MTGCard(6))
    private val genericCards = Arrays.asList(MTGCard(8), MTGCard(9))


    @Before
    fun setup() {
        `when`(set.name).thenReturn("Zendikar")
        cardsSetBucket = CardsBucket(set, setCards)
        genericBucket = CardsBucket("fav", genericCards)
    }

    @Test
    @Throws(Exception::class)
    fun testGetCards() {
        var cards = cardsSetBucket.cards
        assertThat(cards, `is`(setCards))
        cards = genericBucket.cards
        assertThat(cards, `is`(genericCards))
    }

    @Test
    @Throws(Exception::class)
    fun testIsValid() {
        assertTrue(cardsSetBucket.isValid(set.name))
        assertFalse(cardsSetBucket.isValid("fav"))
        assertFalse(genericBucket.isValid(set.name))
        assertTrue(genericBucket.isValid("fav"))
    }

    @Test
    @Throws(Exception::class)
    fun testSetCards() {
        cardsSetBucket.cards = genericCards
        assertThat(cardsSetBucket.cards, `is`(genericCards))
    }

    @Test
    @Throws(Exception::class)
    fun testGetKey() {
        assertThat(cardsSetBucket.key, `is`(set.name))
        assertThat(genericBucket.key, `is`("fav"))
    }
}